### PR TITLE
Introduce ui-manager to render layout on demand

### PR DESCRIPTION
**Pull request by nickhsine**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,10 +1,63 @@
...
+  background-color: ${props => props.backgroundColor};
+`
+
+const ContentBlock = styled.div`
+  position: relative;
+`
+
+class AppShell extends React.PureComponent {
+  static propTypes = {
+    // Below props are provided by Redux
```
Comment (YuCJ) [inline src/app.js]: The comment can be removed

**Comment by nickhsine** [comment]
@YuCJ 
Please help to review this PR.
I refactor the codes again.

File: src/app.js [hunk 2]
```diff
@@ -1,10 +1,152 @@
...
+import { Provider } from 'react-redux'
 import { Switch, Route } from 'react-router-dom'
 import { colors, lineHeight, typography } from './themes/common-variables'
 import { createGlobalStyle } from 'styled-components'
 
+const _ = {
+  values
+}
+
+const AppShellBox = styled.div``
```
Comment (YuCJ) [inline src/app.js]: `AppShellBox` can be replaced with simple `<div>`

File: src/app.js [hunk 3]
```diff
@@ -1,10 +1,152 @@
...
+
+  static defaultProps = {
+    layoutObj: {
+      header: uiConst.header.default,
+      footer: uiConst.footer.default
+    }
+  }
+
+  renderHeader() {
+    const { layoutObj = {}, releaseBranch } = this.props
```
Comment (YuCJ) [inline src/app.js]: Use `static defaultProps` instead?

File: src/app.js [hunk 4]
```diff
@@ -1,10 +1,152 @@
...
+            releaseBranch={releaseBranch}
+            isLinkExternal={false}
+          />
+        )
+      }
+    }
+  }
+
+  renderFooter() {
+    const { layoutObj = {} } = this.props
```
Comment (YuCJ) [inline src/app.js]: Use `static defaultProps` instead?

**Pull request merged**